internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -336,7 +354,53 @@ func (r *BMACReconciler) handleBMHFinalizer(ctx context.Context, log logrus.Fiel
 		return reconcileComplete{stop: true, dirty: true}
 	}
 
-	if dirty := r.configureBMHForCleaning(ctx, log, bmh); dirty {
+	drainTimedOut, err := nodeDrainTimedout(bmh)
```

Comment:
Consider extracting all this to another function (`handleDrain`)

Reply:
Done :+1: